### PR TITLE
Add Level events

### PR DESCRIPTION
**Pull request by Pandetthe**

### CODE REVIEW

**Comment by Pandetthe** [comment]
Added

**Comment by rube200** [comment]
The point wasnt to remove all the events.
More like
UnturnedPreLevelEvent : UnturnedLevelEvent : Event

**Comment by Pandetthe** [comment]
I have skipped Level.onLevelExited, I don't know if it's called serverside

File: unturned/OpenMod.Unturned/Level/Events/LevelEventsListener.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    internal class LevelEventsListener : UnturnedEventsListener
+    {
+        public LevelEventsListener(IServiceProvider serviceProvider) : base(serviceProvider)
+        {
+
+        }
+
+        public override void Subscribe()
+        {
+            SDG.Unturned.Level.onLevelsRefreshed += OnLevelsRefreshed;
```
Comment (DiFFoZ) [inline unturned/OpenMod.Unturned/Level/Events/LevelEventsListener.cs:17]: Remove it, it's not used on server-side

File: unturned/OpenMod.Unturned/Level/Events/LevelEventsListener.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            SDG.Unturned.Level.onPostLevelLoaded += OnPostLevelLoaded;
+            SDG.Unturned.Level.onPreLevelLoaded += OnPreLevelLoaded;
+            SDG.Unturned.Level.onPrePreLevelLoaded += OnPrePreLevelLoaded;
+            SDG.Unturned.Level.onLevelLoaded += OnLevelLoaded;
+        }
+
+        public override void Unsubscribe()
+        {
+            SDG.Unturned.Level.onLevelsRefreshed -= OnLevelsRefreshed;
```
Comment (DiFFoZ) [inline unturned/OpenMod.Unturned/Level/Events/LevelEventsListener.cs:26]: Remove it, it's not used on server-side

File: unturned/OpenMod.Unturned/Level/Events/UnturnedLevelEvent.cs
Comment (DiFFoZ) [inline unturned/OpenMod.Unturned/Level/Events/UnturnedLevelEvent.cs:1]: Do we really need to pass the scene index? I think we only need `Level.BUILD_INDEX_GAME` and everything else should be skipped.

**Review by DiFFoZ — CHANGES_REQUESTED** [review]

File: unturned/OpenMod.Unturned/Level/Events/UnturnedLevelEvent.cs
Comment (Pandetthe) [reply unturned/OpenMod.Unturned/Level/Events/UnturnedLevelEvent.cs:1]: Not really

**Review by DiFFoZ — APPROVED** [review]

**Pull request merged**